src/library/compiler/ir_interpreter.cpp

```diff
@@ -296,7 +296,10 @@ void print_value(tout const & ios, value const & v, type t) {
 }
 
 void * lookup_symbol_in_cur_exe(char const * sym) {
-#ifdef LEAN_WINDOWS
+#if defined(LEAN_EMSCRIPTEN)
+    void* self = dlopen(NULL, RTLD_LAZY);
+    return dlsym(self, sym);
```

Comment:
I'm puzzled by this; it look like the original code should work just fine, given it runs:

https://github.com/emscripten-core/emscripten/blob/0be5b609412d5b07f8157528df8e5088dae84858/system/lib/libc/dynlink.c#L664-L668

which shouldn't care whether you first run `dlopen`:

https://github.com/emscripten-core/emscripten/blob/0be5b609412d5b07f8157528df8e5088dae84858/system/lib/libc/dynlink.c#L543C1-L548C4

Reply:
It looks to me like this is no longer necessary after https://github.com/emscripten-core/emscripten/commit/c4f7a55347044af6ae29c73cec6e054cc3d83d97; what version of empscripten are we targeting?